File: BOJ/자료구조/문자열/2810/2810.py
# 2810 컵홀더
# https://www.acmicpc.net/problem/2810
# 문자열?

N = int(input())
sit=input()
cnt=0
sCnt=sit.count('S')
lCnt=sit.count('LL')

if lCnt>0:
    print(sCnt+lCnt+1)
else:
    print(N)

File: BOJ/자료구조/문자열/2810/2810.cpp
// 2810 컵홀더
// https://www.acmicpc.net/problem/2810
#include <iostream>

using namespace std;

int main(void) {
    string sit;
    int N, sCnt = 0, lCnt =0;

    cin >> N;
    cin >> sit;

    for (char& ch: sit)
    {
        if (ch == 'S') sCnt++;
        if (ch == 'L') lCnt++;
    }

    if (lCnt && lCnt % 2 == 0)
    {
        lCnt /= 2;
        lCnt++;
    }

    cout << lCnt+sCnt << '\n';
    return 0;
}